R/format.R

```diff
@@ -103,6 +103,21 @@ formatSignif = function(
 #'   \code{params = list('ko-KR', list(year = 'numeric', month = 'long', day =
 #'   'numeric'))}
 formatDate = function(table, columns, method = 'toDateString', params = NULL) {
+  x = table$x
+  if (x$filter != 'none') {
+    if (inherits(columns, 'formula')) columns = all.vars(columns)
+    colnames = base::attr(x, 'colnames', exact = TRUE)
+    rownames = base::attr(x, 'rownames', exact = TRUE)
+    if (is.null(params)) params = list()
+    cols = as.character(name2int(columns, colnames, rownames))
```

Comment:
The `as.character()` trick is very subtle here. It relies on the fact of how JS indexes objects: for an object `x` and an integer index `i`, `x[i]` indexes the element with the character name `i`, instead of the `i`-th element. That is how the line 528 works: https://github.com/rstudio/DT/pull/522/files#diff-2c60c8a789627a2716b5b4271c2b398bR528

This trick can be fragile when the number of columns is larger than 100000, because `as.character(100000)` gives you `"1e+5"`. But I imagine it should be extremely rare that users have more than 100 thousand columns in a table.

Reply:
Ha, I didn't notice this. Using `sprintf("%d", 100000)` should be robust enough.